run/authentication/auth.py

```diff
@@ -0,0 +1,39 @@
+# Copyright 2020 Google, LLC.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# [START run_service_to_service_auth]
+import urllib
+
+import google.auth.transport.requests
+import google.oauth2.id_token
+
+
+def make_get_request(service_url):
+    """
+    make_get_request makes a GET request to the specified Cloud Run endpoint in
+    service_url (must be a complete URL) by authenticating with the ID token
+    obtained from the google-auth client library.
+    """
+
+    req = urllib.request.Request(service_url)
+
+    credentials, project = google.auth.default()
```

Comment:
I think this call can be removed. `fetch_id_token` will check the environment for credentials https://github.com/googleapis/google-auth-library-python/blob/c05b8b52e3bbc096cf32e2d4bb5bd45986d3cd04/google/oauth2/id_token.py#L187-L194

>     1. If the application is running in Compute Engine, App Engine or Cloud Run,
>        then the ID token are obtained from the metadata server.
>     2. If the environment variable ``GOOGLE_APPLICATION_CREDENTIALS`` is set
>        to the path of a valid service account JSON file, then ID token is
>        acquired using this service account credentials.
>     3. If metadata server doesn't exist and no valid service account credentials
>        are found, :class:`~google.auth.exceptions.DefaultCredentialsError` will
>        be raised.

Reply:
Thank you!